src/core/Tw2AnimationController.js

```diff
@@ -448,40 +492,138 @@ Tw2AnimationController.prototype.PlayAnimation = function(name, cycle, callback)
-            }
+            animation.cycle = cycle;
+        }
+        if (typeof(callback) != 'undefined')
+        {
+            animation.callback = callback;
         }
+
+        return true;
     }
+
 };
 
 /**
- * Stops a specific animation from playing
- * @param {string} name - Animation Name
+ * Plays a specific animation from a specific time
+ * @param {string} name - Animation's Name
+ * @param {number} from - Time to play from
+ * @param {boolean} [cycle]
+ * @param {Function} [callback] - Optional callback which is fired once the animation has completed
+ * @returns {boolean}
  * @prototype
  */
-Tw2AnimationController.prototype.StopAnimation = function(name)
+Tw2AnimationController.prototype.PlayAnimationFrom = function(name, from, cycle, callback)
 {
-    for (var i = 0; i < this.animations.length; ++i)
+    if (this.animations.length == 0)
     {
-        if (this.animations[i].animationRes.name == name)
+        if (!this.pendingCommands)
+        {
+            this.pendingCommands = [];
+        }
+        this.pendingCommands.push(
+        {
+            'func': this.PlayAnimationFrom,
+            'args': [name, from, cycle, callback]
+        });
+        return true;
+    }
+
+    var animation = this.GetAnimation(name);
+
+    if (animation)
+    {
+        from = (from <= animation.animationRes.duration) ? from : animation.animationRes.duration;
+        animation.time = (from < 0) ? 0 : from;
+        animation.isPlaying = true;
+        if (typeof(cycle) != 'undefined')
+        {
+            animation.cycle = cycle;
+        }
+        if (typeof(callback) != 'undefined')
+        {
+            animation.callback = callback;
+        }
+
+        return true;
+    }
+};
+
+/**
+ * Gets an array of all the currently playing animations by name
+ * @returns {[]}
+ * @constructor
+ */
+Tw2AnimationController.prototype.GetPlayingAnimations = function()
+{
+    var result = [];
+
+    for (var i = 0; i < this.animations.length; i++)
+    {
+        if (this.animations[i].isPlaying)
         {
-            this.animations[i].isPlaying = false;
+            result.push(this.animations[i].animationRes.name)
         }
     }
+
+    return result;
+};
+
+/**
+ * Stops an animation or an array of animations from playing
+ * @param {String, Array.<string>} names - Animation Name, or Array of Animation Names
+ * @return {boolean}
+ * @prototype
+ */
+Tw2AnimationController.prototype.StopAnimation = function(names)
+{
+    if (this.animations.length == 0)
+    {
+        if (!this.pendingCommands)
+        {
+            this.pendingCommands = [];
+        }
+        this.pendingCommands.push(
+        {
+            'func': this.StopAnimation,
+            'args': names
+        });
+        return true;
+    }
+
+    if (typeof names == 'string' || names instanceof String)
+    {
+        names = [names];
+    }
+
+    if (names && Object.prototype.toString.apply(names) === '[object Array]')
```

Comment:
If "names" is either string or array of strings and you already handled string case, can't you just assume that names is an array at this point?